### ISSUE TITLE
PowerPoint reader support?

### ISSUE THREAD

**Issue by kaisbe** [jgm/pandoc#4252]
Hi,

I am using pandoc v2.1. Unfortunately converting from PowerPoint format doesn't seem to be supported - this is the output generated by `pandoc --list-input-formats` command:

> commonmark
> creole
> docbook
> docx
> epub
> gfm
> haddock
> html
> jats
> json
> latex
> markdown
> markdown_github
> markdown_mmd
> markdown_phpextra
> markdown_strict
> mediawiki
> muse
> native
> odt
> opml
> org
> rst
> t2t
> textile
> tikiwiki
> twiki
> vimwiki

**Comment by jkr** [comment]
PowerPoint would be *very* difficult to convert from. Like PDF, it's a visual format: the order in the xml is in terms of the stack of shapes on a slide, but that doesn't correspond to where they sit on the screen. (You can have the lowest level one on either the top or the bottom, or both depending on the slide theme.) There would be no dependable way to transfer it into a meaningful structured document.

**Comment by kaisbe** [comment]
@jkr Ok, understood. Thanks anyway.

**Comment by sisrfeng** [comment]
> the order in the xml is in terms of the stack of shapes on a slide
>  no dependable way to transfer it into a meaningful structured document.
Many pptx files do not have stacking shapes or elements. I think it may be helpful if pandoc supports pptx reader and output as much as possible.

**Comment by tibbsa** [comment]
@sisrfeng Not a solution to bringing this into pandoc, but I just ran into this need as well and this might be of help.  It assumes a LOT, but if you're just looking to pull the titles/bullets out of a PPT into a Word document, it might get you there.

https://gist.github.com/tibbsa/41f0ee4400ef49ce1b14960eab8c8788

**Comment by allefeld** [comment]
There is [pptx2md](https://github.com/ssine/pptx2md), a Python script to convert `pptx` to Markdown. It has a `--qmd` option for output intended to be used with Quarto, which is basically a wrapper around Pandoc. It works decently, but the markdown needs a lot of manual editing.

**Comment by chrispy-snps** [comment]
We use [pptx2md](https://github.com/ssine/pptx2md) and it does a reasonably good job of getting the gist of the content. Bullet structures come through (with formatting), and its heuristic of collapsing slide titles across same-titled slides works well.

Even if the conversion from .pptx into Pandoc is lossy, it would still be extremely useful. Every other method we're forced to use for PowerPoint conversion is also lossy. :)

**Comment by fcakyon** [comment]
Having at least a limited support for pptx input (for simple presentations) in `pandoc` would be very useful for our use case 👍🏻

**Comment by allefeld** [comment]
In the meantime, I found that pptx2md chokes on some pptx files.

Looking for a fix, I found this relatively new project: [microsoft/markitdown](https://github.com/microsoft/markitdown), a Python library & cli tool.

I know, somewhat weird seeing MS providing software to convert its mess into something readable. Their motivation is processing by LLMs, which like Markdown. 

I just had a quick look, so I'm not sure how faithful the conversion is. The output is not pretty, but nothing which a run through Pandoc can't fix.

Both pptx2md and markitdown are based on the existing Python library python-pptx.
